jax/interpreters/mlir.py

```diff
@@ -542,6 +542,7 @@ def register_lowering(prim: core.Primitive, rule: LoweringRule,
     # this expansion.
     for p in xb.expand_platform_alias(platform):
       _platform_specific_lowerings[p][prim] = rule
+  return rule
```

Comment:
Why do you need this return rule?

shard_map.py doesn't do anything with the returned value AIUI?

Reply:
That change is unrelated, and could be removed from this PR. It's just a good pattern when you might use something as a decorator, like

```python
@register_thing
def foo(...):
  ...
```

If `register_thing` doesn't return its argument then the code above binds `foo = None`.

Reply:
Ohh, cool! Thanks for explaining.

Reply:
I don't remember what made me notice it while working on this... but maybe at one point I wanted to write

```python
@partial(mlir.register_lowering, shard_map_p)
def _shard_map_lowering(...):
  ...
```

and still have `_shard_map_lowering` bound to the actual function object afterwards.